akka-cluster-typed/src/main/scala/akka/cluster/ddata/typed/scaladsl/Topic.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.ddata.typed.scaladsl
+import akka.actor.typed.receptionist.{Receptionist, ServiceKey}
+import akka.actor.typed.{ActorRef, Behavior}
+import akka.actor.typed.scaladsl.Behaviors
+import akka.annotation.DoNotInherit
+
+object Topic {
+
+  /**
+   * NOT FOR USER EXTENSION
+   */
+  @DoNotInherit
+  sealed trait TopicCommand
+
+}
+
+class Topic[T](serviceKey: ServiceKey[T]) {
+  import Topic._
+
+  final case class Subscribe(subscriber: ActorRef[T]) extends TopicCommand
+  final case class Unsubscribe(subscriber: ActorRef[T]) extends TopicCommand
+  final case class Publish(msg: T) extends TopicCommand
+
+  private case class RemoteSubscribersChanged(subscriber: Set[ActorRef[T]]) extends TopicCommand
+  private case class WasPublished(t: T) extends TopicCommand
+
+  def behavior[T]: Behavior[TopicCommand] = noSubscribersBehavior
+
+
+  private def noSubscribersBehavior: Behavior[TopicCommand] = Behaviors.receiveMessage {
+    case Subscribe(subscriber) =>
+      Behaviors.setup { ctx =>
+        val remoteMessageAdapter = ctx.messageAdapter {
+          case t: T @unchecked => WasPublished(t)
+        }
+        ctx.system.receptionist ! Receptionist.register(serviceKey, remoteMessageAdapter)
```

Comment:
why use Java API here?

Reply:
Coding-while-also-listening-to-talk-accident